### PR TITLE
`basic-authorship`: debug level is now less spammy

### PR DESCRIPTION
**Pull request by michalkucharczyk**
The `debug` level in `sc-basic-authorship`  is now less spammy. Previously it was outputing logs per individual transactions. It was quite hard to follow the logs (and also generates unneeded traffic in grafana).

Now `debug` level only show some internal details, without spamming output with per-transaction logs. They were moved to `trace` level.

I also added the `EndProposingReason` to the summary INFO message. This allows us to know what was the block limit (which is very useful for debugging).

Example:
```
🎁 Prepared block for proposing at 64 (1186 ms) hash: 0x4b5386c13c507d0dbab319ac054cc1bcfa08311e184452221ad07f12ecc6091c; parent_hash: 0x157c…ca5e; end: HitBlockWeightLimit; extrinsics_count: 7032; 
```

### CODE REVIEW

File: substrate/client/basic-authorship/src/basic_authorship.rs [hunk 1]
```diff
@@ -565,20 +565,22 @@ where
 
 		if log::log_enabled!(log::Level::Info) {
 			info!(
-				"🎁 Prepared block for proposing at {} ({} ms) [hash: {:?}; parent_hash: {}; extrinsics_count: {}",
+				"🎁 Prepared block for proposing at {} ({} ms) hash: {:?}; parent_hash: {}; end: {:?}; extrinsics_count: {};",
```
Comment (lexnv) [inline substrate/client/basic-authorship/src/basic_authorship.rs:568]: nit: Do we need ; at the end?

**Review by lexnv — APPROVED** [review]

File: substrate/client/basic-authorship/src/basic_authorship.rs [hunk 2]
```diff
@@ -565,20 +565,22 @@ where
...
 				block.header().number(),
 				block_took.as_millis(),
 				<Block as BlockT>::Hash::from(block.header().hash()),
 				block.header().parent_hash(),
+				end_reason,
```
Comment (bkchr) [inline substrate/client/basic-authorship/src/basic_authorship.rs:573]:
```suggestion
				"🎁 Prepared block for proposing at {} ({} ms) hash: {:?}; parent_hash: {}; extrinsics_count: {};",
				block.header().number(),
				block_took.as_millis(),
				<Block as BlockT>::Hash::from(block.header().hash()),
				block.header().parent_hash(),
```

No need to add log this in info.

**Review by bkchr — APPROVED** [review]

**Review by skunert — APPROVED** [review]

File: substrate/client/basic-authorship/src/basic_authorship.rs (hunk 2 above)
Comment (michalkucharczyk) [reply substrate/client/basic-authorship/src/basic_authorship.rs:573]: I need it here, this is why I added it. I need to know why block builder is hitting limits. And we may have it "for free" in this line.

I am now confused, it is OK to merge it with end reason or not?

File: substrate/client/basic-authorship/src/basic_authorship.rs (hunk 1 above)
Comment (michalkucharczyk) [reply substrate/client/basic-authorship/src/basic_authorship.rs:568]:
```suggestion
				"🎁 Prepared block for proposing at {} ({} ms) hash: {:?}; parent_hash: {}; end: {:?}; extrinsics_count: {}",
```

File: substrate/client/basic-authorship/src/basic_authorship.rs (hunk 2 above)
Comment (bkchr) [reply substrate/client/basic-authorship/src/basic_authorship.rs]: Why do you need it there? You could just enable trace logging.

Comment (bkchr) [reply substrate/client/basic-authorship/src/basic_authorship.rs]: Also we have metrics for this.

Comment (skunert) [reply substrate/client/basic-authorship/src/basic_authorship.rs]: I think its useful to have, with tracing it will spam all tx again.

Comment (michalkucharczyk) [reply substrate/client/basic-authorship/src/basic_authorship.rs]: I don't want to enable trace, because it is way too spammy. With 7k txs in block it may affect the block time building. And for sure it does not help with collecting logs in grafana/loki - it sometimes breaks with too many logs.

I can add another "if branch" here for debug, but I think it will more confusing when reasoning about what to enable.

Comment (michalkucharczyk) [reply substrate/client/basic-authorship/src/basic_authorship.rs]: It is not one network, instead of having important info in log I always need to restart the node whiat is a cost (time). having this log allows very quick assessment of what is happening in block builder. 

and logs are not only for operator. after all why does operator need this hash?

I would not propose it if I could live w/o this. And finally @skunert says he finds it useful too.
